README.md

```diff
@@ -51,82 +65,155 @@ The environment variable `DEBUG_ERROR` (boolean) controls whether or not errors
-
-`REDUX_LOG` and `REDUX_DEV_UI` are both turned on by default in `config/ui-debug.sh`.
+`REDUX_LOG` is turned on by default in `config/ui-debug.sh`.
 
 ### Local Development w/o Debug Mode(s)
 
 All debug options are turned *off* by default in `config/local.sh`.
 
 ## Tests
 
-There are two sets of (unit) tests for the code in this repository.
-
-The tests for all device and data-processing code currently run in the [nodejs](https://nodejs.org/en/) server-side JavaScript environment. (We plan to eventually migrate these tests to run in-browser since the code itself runs in-browser in the Chrome App.)
-
-The tests for all the UI code run using the [Karma test runner](https://karma-runner.github.io/0.13/index.html) in [the headless WebKit browser PhantomJS](http://phantomjs.org/) or the Chrome browser.
-
-To run the tests in this repository as they are run on Travis CI, use:
+To run the tests in this repository as they are run on CircleCI and Appveyor use:
 
 ```bash
 $ npm test
 ```
+or
+```bash
+$ yarn test
+```
 
-To run just the UI tests in both PhantomJS and Chrome *locally*, use:
+## Linting & Code Style
 
-```bash
-$ npm run browser-tests
+We use [ESLint](http://eslint.org/) to lint our JavaScript code. We try to use the same linting options across all our client apps, but there are a few exceptions in this application, noted with comments in the `.eslintrc` configuration file.
+
+To run the linter (which also runs on CircleCI and Appveyor with every push, along with `npm test`), use:
+
+```
+$ npm run lint
 ```
 
-To run just the device and data-processing tests in node, use:
+Aside from the (fairly minimal) JavaScript code style options we *enforce* through the linter, we ask that internal developers and external contributors try to match the style of the code in each module being modified. New modules should look to similar modules for style guidance. In React component code, use existing ES6/ES2015 components (not legacy ES5 components) as the style model for new components.
+
+**NB: Please keep ES5 and ES6/ES2015 code distinct. Do *NOT* use ES6/ES2105 features in ES5 modules (most easily recognizable by the use of `require` rather than `import`).**
+
+## Docs
+
+Docs reside in several places in this repository, such as `docs/` and `lib/drivers/docs`. They are built as a static site with [GitBook](https://www.gitbook.com/ 'GitBook') and served at [developer.tidepool.io/](http://developer.tidepool.io/) via [GitHub Pages](https://pages.github.com/ 'GitHub Pages').
+
+See [this guidance on our use of GitBook at Tidepool](http://developer.tidepool.io/docs/).
+
+## Publishing
+
+Release management and application updates are handled via the Github provider in the `electron-builder` project. The recommended workflow for a new production release is as follows:
+
+1. When you're working on what might become a new release, increment the version number in `package.json` and `app/package.json` and commit/push (on the branch)
```

Comment:
We could use it, but we also don't want to create tags locally and push them to GitHub in this flow (necessarily) - I've been using the GitHub releases mechanism to create the tag for me (the flow recommended by `electron-builder`), as actually making the release on GitHub will prevent new build artifacts from being pushed to that version (as matched in the `app/package.json` file it seems.) 

`npm version` can be passed any valid `semver` string, so we could use it for `alpha` versions or just use `mversion` inside the `app` directory (and without the `-m` flag, to prevent it from making a tag). 

It is slightly strange to only update the `app/package.json` as I almost always expect to find the current version of an application in the root `package.json` file. 

I don't have strong feelings about an approach so if `npm version` is preferable to you, we can work that in.